requirements-dev.txt

```diff
@@ -2,3 +2,7 @@ coverage==4.4.1
 nose==1.3.7
 pylint==1.7.2
 tox==2.8.2
+twine==1.12.1
```

Comment:
I haven't used twine in the past. Is it required for fixing this issue? Since this requirement is only for pushing to pypi, wouldn't it be better to let end developer decide how he/she wishes to publish package?

Reply:
Thanks @manrajgrover!

According to Pypa,  `twine` is now the [official way of publishing python packages to Pypi](https://packaging.python.org/tutorials/packaging-projects/).  The `twine upload` command handles both registering the package name and uploading the distributions.

I typically have a Makefile in my python projects with the following commands:
```
## remove all build, test, coverage and Python artifacts
clean: clean-build clean-pyc clean-test

## remove build artifacts
clean-build:
	rm -fr build/
	rm -fr dist/
	rm -fr .eggs/
	find . -name '*.egg' -exec rm -f {} +

## remove Python file artifacts
clean-pyc:
	find . -name '*.pyc' -exec rm -f {} +
	find . -name '*.pyo' -exec rm -f {} +
	find . -name '*~' -exec rm -f {} +
	find . -name '__pycache__' -exec rm -fr {} +
	rm -rf .mypy_cache/

## remove test and coverage artifacts
clean-test:
	rm -fr .tox/
	rm -f .coverage
	rm -fr htmlcov/
	rm -fr .cache/
	rm -fr .pytest_cache

## builds source and wheel package
dist: clean
	python setup.py sdist bdist_wheel


## upload wheel and source distribution
upload: dist
	twine upload -r pypi dist/hexpy*

## builds source and wheel package
dist: clean
	python setup.py sdist bdist_wheel

```

With a `.pypirc` file in my home directory i can run `make upload` which will remove all previous artifacts, make a fresh build in the `dist/` directory and upload them to Pypi.

`twine upload -s` will also let you sign your distribution with GPG.

Reply:
Also just noticed this in the tutorial I linked to from the original PR.
> You must upload the distribution with twine, and only twine>=1.11.0 will correctly send the metadata to PyPI: